src/Serializer/SubjectsAware.php

```diff
@@ -19,8 +20,11 @@
         $this->subjects = $subjects;
     }
 
-    final private function getSubjects(array $subjects) : PromiseInterface
+    final private function getSubjects($subjects) : PromiseInterface
```

Comment:
Unfortunately the `object -> json -> object` still uses ArrayCollection interchangeably